### PR TITLE
Feature/bluemap api v2

### PR DESCRIPTION
**Pull request by Aurelien30000**
Support for BlueMap v3 ~ BlueMapApi v2
Switch to Java 11

_Sorry for the massive code refactor, I honestly noticed too inconsistencies in the code format so I decided to let Intellij fix thoses. You may ignore the related commit, it is not breaking at all._

Tested with the WorldGuard integration, expected behavior for now. Hopefully I did not missed something or messed up the whole code :D

### CODE REVIEW

**Comment by Mark-225** [comment]
Thank you so much for not only porting to v2 but also cleaning up after me! I honestly forgot how messy this code was xD  
I will probably still change something about the UpdateTask before the official release, because it doesn't need to be that async anymore since API v2 doesn't require IO operations. But once I have fully reviewed and tested your code I will just release this as a snapshot for the people who need v2 right now.  
Again very much appreciated :)

File: BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java [hunk 1]
```diff
@@ -3,102 +3,86 @@
...
+    public void onEnable(BlueMapAPI blueMapAPI){
         this.blueMapAPI = blueMapAPI;
         Bukkit.getScheduler().runTask(BlueBridgeCore.getInstance(), () ->{
             BlueBridgeCore.getInstance().updateConfig();
             BlueBridgeCore.getInstance().reloadAddons();
             resetMarkers();
             UpdateTask.worlds.clear();
             UpdateTask.resetLastSnapshots();
-            UpdateTask.worlds.addAll(blueMapAPI.getWorlds().stream().map(BlueMapWorld::getUuid).collect(Collectors.toList()));
+            UpdateTask.worlds.addAll(blueMapAPI.getWorlds().stream().map(blueMapWorld -> UUID.fromString(blueMapWorld.getId())).collect(Collectors.toList()));
```
Comment (TBlueF) [inline BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java:36]: You can't rely on `BlueMapWorld#getId()` being a UUID .. it is in most cases, but it's very well possible to be something else theoretically

File: BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/config/BlueBridgeConfig.java [hunk 2]
```diff
@@ -1,63 +1,63 @@
...
         return config.getDouble("maxDistance", 500.0);
     }
 
-    public static synchronized Color defaultColor(){
+    public static synchronized Color defaultColor() {
         String rgba = config.getString("defaultColor", "960087ff");
-        if(!rgbaRegex.matcher(rgba).matches())
+        if (!rgbaRegex.matcher(rgba).matches())
             rgba = "960087ff";
         return BlueBridgeUtils.stringToColor(rgba);
```
Comment (TBlueF) [inline BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/config/BlueBridgeConfig.java:50]: Could be replaced with `new Color("#" + rgba)` =)

**Review by TBlueF — COMMENTED** [review]
Couldn't resist :3

File: BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java (hunk 1 above)
Comment (Aurelien30000) [reply BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java:36]: What else on Bukkit :thinking:

Comment (Mark-225) [reply BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java:36]: It is better to map it the other way around. Instead of trying to get the Bukkit World UUID from a BlueMapWorld, try to get the BlueMap World for each loaded Bukkit World and collect the ones which are present. Then save them in a HashMap mapping UUIDs to BlueMapWorld Objects if you need them later.  
If you call `BlueMapAPI.getWorld(uuid)` BlueMap will reliably return the BlueMapWorld for that Bukkit World UUID. But the other way around it might change at some point since the API does not guarantee you to return a UUID String on `BlueMapWorld.getId()`.  
So basically both approaches work fine, but this might not be as future-proof as the alternative.

File: BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/config/BlueBridgeConfig.java (hunk 2 above)
Comment (Aurelien30000) [reply BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/config/BlueBridgeConfig.java:50]: _This requires some internal changes ``AARRGGBB`` -> ``RRGGBBAA``._

File: BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java (hunk 1 above)
Comment (Aurelien30000) [reply BlueBridgeCore/src/main/java/de/mark225/bluebridge/core/bluemap/BlueMapIntegration.java:36]: Thanks to both of you!

**Comment by Aurelien30000** [comment]
Tell me if this sounds better for ids.

**Comment by Mark-225** [comment]
Merging this into main for now, release will follow when I properly merged all the dev branch changes that happened in the mean time.
